core/src/test/java/org/pentaho/di/connections/utils/ConnectionUriParserTest.java

```diff
@@ -0,0 +1,184 @@
+
+    assertNullValues( new ConnectionUriParser( "someGarbage" ) );
+
+  }
+
+  @Test
+  public void testConnectionUriParser_Negative_Example_Non_URIs() throws Exception {
+
+    assertNullValues( new ConnectionUriParser( "/someUser/someUnixFile" ) );
+
+    assertNullValues( new ConnectionUriParser(  "T:\\Users\\RandomSUser\\Documents\\someWindowsFile" ) );
+
+    assertNullValues( new ConnectionUriParser(  "//home/randomUser/randomFile.rpt" ) ); // Pentaho repository
+
+  }
+
+  @Test
+  public void testConnectionUriParser_Example_URIs() throws Exception {
+
+    String uri_01 = "xyz://";
+    assertEquals( "xyz", new ConnectionUriParser( uri_01 ) );
+
+    String uri_02 = "xyz:///";
+    assertEquals( "xyz", new ConnectionUriParser( uri_02 ) );
+
+    String uri_03 = "pvfs://";
+    assertEquals( "pvfs", new ConnectionUriParser( uri_03 ) );
+
+    String uri_04 = "pvfs:///";
+    assertEquals( "pvfs", new ConnectionUriParser( uri_04 ) );
+
+    String uri_05 = "xyz://abc";
+    assertEquals( "xyz", "abc", new ConnectionUriParser( uri_05 ) );
+
+    String uri_06 = "xyz://abc/";
+    assertEquals( "xyz", "abc", new ConnectionUriParser( uri_06 ) );
+
+    String uri_07 = "xyz://abc/someDir/somePath/someFile.txt";
+    assertEquals( "xyz",
+      "abc",
+      "/someDir/somePath/someFile.txt",
+      new ConnectionUriParser( uri_07 ) );
+
+    String uri_08 = "pvfs://some-ConnectionName_123/";
+    assertEquals( "pvfs",
+      "some-ConnectionName_123",
+      new ConnectionUriParser( uri_08 ) );
+
+    String uri_09 = "pvfs://some-ConnectionName_123";
+    assertEquals( "pvfs",
+      "some-ConnectionName_123",
+      new ConnectionUriParser( uri_09 ) );
+
+    String uri_10 = "pvfs://some-ConnectionName_123/someFolderA/someFolderB/someFolderC/sales_data.csv";
+    assertEquals( "pvfs",
+        "some-ConnectionName_123",
+        "/someFolderA/someFolderB/someFolderC/sales_data.csv",
+        new ConnectionUriParser( uri_10 ) );
+
+    // TEST : can handle special characters, passing connection name as-is per "current" requirements based on connection creation logic
+    String uri_11 = "pvfs://Special Character name &#! <> why would you do this/someFolderA/someFolderB/someFolderC/sales_data.csv";
+    assertEquals( "pvfs",
+      "Special Character name &#! <> why would you do this",
+      "/someFolderA/someFolderB/someFolderC/sales_data.csv",
+      new ConnectionUriParser( uri_11 ) );
+
+    String uri_12 = "pvfs://Special Character name &#! <> why would you do this";
+    assertEquals( "pvfs",
+      "Special Character name &#! <> why would you do this",
+      new ConnectionUriParser( uri_12 ) );
+
+    String uri_13 = "pvfs://Special Character name &#! <> why would you do this/";
+    assertEquals( "pvfs",
+      "Special Character name &#! <> why would you do this",
+      new ConnectionUriParser( uri_13 ) );
+
+    String specialCharacterConnectionName = shuffleString( ACCEPTED_CHARACTERS_FULL_SET );
```

Comment:
I think having a random test case is not a good idea. I would prefer having several test strings with various deemed relevant cases, directly written in the test class.

Reply:
Overview:

1. No code change on the randomize part, that seems a subjective ask.
2. But I can add a connectionName with international characters ie` "somee_Cool_characters_é"`, that's a miss on my part.
3. As we spoke during scrum, our manager is tracking down what characters set is allowed. So however we document what we "support" either through test cases and/or javadoc will need to be updated. But as you see in the code, I'm literally taking what ever the first segment ie ` ://<connectionName/`. When we get that character set, I'm sure we'll have to update the javadoc, test cases, and hopefully not any source code.

The main goal is that there is a test case the has all the special characters that we want to verify that the class `ConnectionUriParser.java` supports.

That string `ConnectionUriParser#ACCEPTED_CHARACTERS_FULL_SET` has the special characters `ConnectionUriParser#SPECIAL_CHARACTERS_FULL_SET`. `ConnectionUriParser#SPECIAL_CHARACTERS_FULL_SET` with the set of special characters we support as of now, that could change. I'm open to modifications.

Wether the I create a connectionName from

- directly as in from `ConnectionUriParser#ACCEPTED_CHARACTERS_FULL_SET`
- or random string of `ConnectionUriParser#ACCEPTED_CHARACTERS_FULL_SET`
is in my opinion irrelevant, the contents of the string are the same.

Reply:
Addressed in commit: https://github.com/pentaho/pentaho-kettle/commit/57a7b9ed55209fb93c39549b26b39f2a50442921

Reply:
I'm trying to triage all comments for this PR. I've stated my position and you have stated your position very well.

Is this a item that should prevent merging as ?

If so, we should  go offline and get a technical moderator to make a final decision.

Reply:
Resolving. See https://github.com/pentaho/pentaho-kettle/pull/9310#pullrequestreview-2065857215.